### PR TITLE
Making email optional for gift

### PR DESCRIPTION
**Pull request by harshvitra**

### CODE REVIEW

File: src/features/donations/components/treeDonation/GiftForm.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { useForm } from 'react-hook-form';
 import MaterialTextField from '../../../common/InputTypes/MaterialTextField';
 import styles from './../../styles/Donations.module.scss';
 import i18next from '../../../../../i18n';
+import { InputAdornment } from '@material-ui/core';
```
Comment (norbertschuler) [inline src/features/donations/components/treeDonation/GiftForm.tsx:6]: I wonder where this is used?

File: src/features/donations/components/treeDonation/GiftForm.tsx [hunk 2]
```diff
@@ -75,37 +81,64 @@ export default function GiftForm({
...
-              <span className={styles.formErrors}>
-                {t('donate:emailRequired')}
-              </span>
-            )}
+              <MaterialTextField
+                multiline
+                rowsMax="4"
+                label={t('donate:giftMessage')}
+                variant="outlined"
+                name={'giftMessage'}
```
Comment (norbertschuler) [inline src/features/donations/components/treeDonation/GiftForm.tsx:104]: The input given for the gift message always gets lost when navigating back for this form if the gift email was entered and also if no gift email was specified:

Bildschirmfoto 2021-06-15 um 09 42 28

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Comment by norbertschuler** [comment]
- [ ] Not giving an email for a gift donation results into this view in the backend (develop):

Bildschirmfoto 2021-06-15 um 09 51 50

**Comment by norbertschuler** [comment]
> * [ ]  Not giving an email for a gift donation results into this view in the backend (develop) for the gift list and also in the donation details view, e.g.:

This is working now:

Bildschirmfoto 2021-07-20 um 14 00 07
Bildschirmfoto 2021-07-20 um 14 00 39

File: src/features/donations/components/treeDonation/GiftForm.tsx (hunk 2 above)
Comment (norbertschuler) [reply src/features/donations/components/treeDonation/GiftForm.tsx:104]: @harshvitra maybe if you have time you can work on this to get this pull request finished?

**Comment by sarveshpro** [comment]
Donations will be replaced by `planet-donations`

**Pull request closed without merging**